--- a/mainwindow.h	2015-06-09 15:44:52.965202266 +0900
+++ b/mainwindow.h	2017-09-07 14:33:35.261071585 +0900
@@ -50,6 +50,36 @@ public:
     bool GameInit();
     bool GameStart();
 
+    /*======Autoware Socket=================*/ 
+
+    // actuation
+    void SteeringControl(double current_steering_angle, double cmd_steering_angle);
+    void StrokeControl(double current_velocity, double cmd_velocity);
+    void VelocityControl(double current_velocity, double cmd_velocity);
+
+    // cmd
+    static void* CMDGetterEntry(void *a);
+
+    // mode
+    static void* ModeSetterEntry(void *a);
+    void ModeSet(int mode);
+    void SetStrMode(int mode);
+    void SetDrvMode(int mode);
+
+    // gear
+    static void* GearSetterEntry(void *a);
+    void SetGear(int gear);
+
+    // can
+    void SendCAN(void);
+
+    // common
+    bool ConfigSocket(void);
+    void UpdateState(void);
+    void ClearCntDiag(void);
+
+    /*====================================*/
+
 protected:
     void changeEvent(QEvent *);
     void closeEvent(QCloseEvent *);
@@ -207,7 +237,7 @@ private:
     void viewErrCode();
     void viewFirmVersion();
     void updateTime();
-    void writeLog();
+    //void writeLog();
 
     void logThread();
     static void* LogThreadEntry(void* arg);
@@ -274,6 +304,10 @@ private:
 
     int _drvTargetVeloc;
     int _drvTargetStroke;
+
+   /*======Autoware Socket=================*/ 
+    pthread_t _cmdgetter;
+    /*====================================*/
 };
 
 #endif // MAINWINDOW_H
