--- a/mainwindow.cpp	2015-06-09 15:44:52.968202279 +0900
+++ b/mainwindow.cpp	2017-09-07 14:34:57.328772739 +0900
@@ -217,6 +217,15 @@ MainWindow::MainWindow(QWidget *parent)
     ui->pushButton_startLog->setEnabled(true);
     ui->pushButton_stopLog->setEnabled(false);
 
+    /////////////////////////////////////////////////////////
+    // Autoware extension
+    if (!ConfigSocket()) {
+      printf("Error: failed to configure Autoware socket!\n");    
+    }
+    pthread_create(&_cmdgetter, NULL, CMDGetterEntry, this);
+    pthread_detach(_cmdgetter);
+    ////////////////////////////////////////////////////////
+
     pthread_create(&_logThread, NULL, LogThreadEntry, this);
     readTimer->start(100);
     drvTimer->start(200);
@@ -252,6 +261,9 @@ MainWindow::MainWindow(QWidget *parent)
 
 MainWindow::~MainWindow()
 {
+    hev->Stop();
+    hev->Close();
+
     delete ui;
 }
 
@@ -289,10 +301,13 @@ void MainWindow::logThread()
 //    readTimer->start(10);
     while(1){
         updateTime();
-        if(_selectLog.start == true){
-            writeLog();
-        }
-        usleep(10*1000);
+	/////////////////////////////////////////////////
+	// Autoware extension
+	if(_selectLog.start == true){
+	  //writeLog();
+	  SendCAN();
+	}
+        //usleep(10*1000);
     }
 }
 
@@ -467,12 +482,14 @@ void MainWindow::closeEvent(QCloseEvent
 void MainWindow::startLog()
 {
     updateTime();
+#ifdef NEVER
     char logFile[32];
     sprintf(logFile, "%d_%d_%d_%d_%d_%d.txt",
             _s_time->tm_year+1900, _s_time->tm_mon+1, _s_time->tm_mday,
             _s_time->tm_hour,_s_time->tm_min, _s_time->tm_sec);
 
     _logSave = fopen(logFile, "w+");
+#endif // NEVER
     ui->pushButton_startLog->setEnabled(false);
     ui->pushButton_stopLog->setEnabled(true);
 
@@ -1152,6 +1169,7 @@ void MainWindow::sndDrvTarget()
     }
 }
 
+#ifdef DO_NOT_USE
 void MainWindow::writeLog()
 {
     fprintf(_logSave,"%d/%d/%d/%d:%d:%d.%ld,",_s_time->tm_year+1900, _s_time->tm_mon+1, _s_time->tm_mday,
@@ -1202,6 +1220,7 @@ void MainWindow::writeLog()
 
     fprintf(_logSave, "\n");
 }
+#endif // DO_NOT_USE
 
 void MainWindow::updateTime()
 {
@@ -1239,6 +1258,10 @@ void MainWindow::updateTime()
     *date += "_" + QString::number(_getTime.tv_usec ,10);
 
     _update = *date;
+    /////////////////////////////////////////////////////////
+    // Autoware patch
+    delete(date); // added by yabuta. may cause memory leak?
+    /////////////////////////////////////////////////////////
 //    ui->textEdit_updateTime->setText(update);
 }
 
